src/clients.rs

```diff
@@ -84,3 +92,29 @@ impl fmt::Debug for ExampleClient {
             .finish()
     }
 }
+
+impl Valuable for ExampleClient {
```

Comment:
I wonder if we should have a trait for service clients (including sub-clients) and a derive macro. Then we could have a blanket impl of `impl<T> Valuable for T where T: azure_core::Client`[^1]. Clients could look something like this:

```rust
#[derive(Client)]
#[azure(namespace = "Microsoft.KeyVault")]
pub struct SecretClient {
}
```

[^1]: Or, if that doesn't work, derive `Valuable` in the derive macro expansion.

Reply:
I was thinking something along those lines last night as well. :smile:

All this might change if we instead use the `opentelemetry` crates. I need to look more into those, but internal conversations are happening and Microsoft at large has been contributing to those much like `symcrypt` we'll use as needed.